small_ms_test_case_baseline.py

```diff
@@ -0,0 +1,452 @@
+bl_dist_time_bins = longest_baseline//bl_uvw_dist
+
+print("bl_uvw_dist \n", bl_uvw_dist, bl_uvw_dist.shape)
+print("longest_baseline", longest_baseline)
+print("bl_uvw_dist_ratio \n", bl_dist_time_bins, bl_dist_time_bins.shape)
+
+
+#time_bin_secs = 6
+ubl, _, bl_inv, _ = unique_baselines(ant1, ant2)
+utime, _, time_inv, _ = unique_time(timestamps)
+
+# indeces to recostruct original arrays 
+for idx in bl_inv:
+    print(ubl[idx])
+for idx in time_inv:
+    print(utime[idx])
+
+nbl = ubl.shape[0]
+ntime = utime.shape[0]
+
+sentinel = np.finfo(timestamps.dtype).max
+out_rows = numba.uint32(0)
+
+
+print("ubl \n", ubl, ubl.shape[0])
+print("bl_inv \n", bl_inv)
+print("utime \n", utime, utime.shape[0])
+print("time_inv \n", time_inv)
+
+scratch = np.full(3*nbl*ntime, -1, dtype=np.int32)
+print("scratch \n", scratch, scratch.shape)
+row_lookup = scratch[:nbl*ntime].reshape(nbl, ntime)
+print("scratch[:nbl*ntime] \n", scratch[:nbl*ntime])
+print("row_lookup \n", row_lookup)
+bin_lookup = scratch[nbl*ntime:2*nbl*ntime].reshape(nbl, ntime)
+print("scratch[nbl*ntime:2*nbl*ntime] \n", scratch[nbl*ntime:2*nbl*ntime])
+print("bin_lookup \n", bin_lookup)
+inv_argsort = scratch[2*nbl*ntime:]
+print("inv_argsort \n", inv_argsort)
+time_lookup = np.zeros((nbl, ntime), dtype=timestamps.dtype)
+print("time_lookup \n", time_lookup)      
+interval_lookup = np.zeros((nbl, ntime), dtype=interval.dtype)
+print("interval_lookup \n", interval_lookup)
+bin_flagged = np.zeros((nbl, ntime), dtype=np.bool_)
+print("bin_flagged \n", bin_flagged)
+
+
+print("bl_inv \n", bl_inv)
+print("time_inv \n", time_inv)
+for r in range(timestamps.shape[0]):
+    bl = bl_inv[r]
+    t = time_inv[r]
+    row_lookup[bl, t] = r
+print("row_lookup \n", row_lookup)
+
+# How Do I use row_lookup for find original values ?
+print("lets see")
+for idx in range(row_lookup.size):
+    print(timestamps[idx], interval[idx])
+    print(ant1[idx], ant2[idx])
+
+print("verify")
+for idx in row_lookup:
+    print(timestamps[idx], interval[idx])
+    print(ant1[idx], ant2[idx])
+print(ant1, ant2)
+    
+print("\n constructing bin_lookup and time_lookup \n")
+# Average times over each baseline and construct the
+# bin_lookup and time_lookup arrays
+for bl in range(ubl.shape[0]):
+    tbin = numba.int32(0)
+    bin_count = numba.int32(0)
+    bin_flag_count = numba.int32(0)
+    bin_low = timestamps.dtype.type(0)
+
+    # range of unique time (utime)
+    for t in range(utime.shape[0]):
+        # Lookup input row
+        r = row_lookup[bl, t]
+        print("bl : ", bl, "t : ", t, "r = ", r)
+        # Ignore if not present
+        if r == -1:
+            continue
+
+            
+#         half_int = interval[r] * 0.5
+#         print("half_int",half_int)
+#         if bin_count == 0:
+#             bin_low = timestamps[r] - half_int
+#             print("bin_count ", bin_count)
+#             print("bin_low", bin_low)
+#         elif timestamps[r] + half_int - bin_low > bl_dist_time_bins[bl]:
```

Comment:
At this point, we're adding samples to our current bin. For time averaging we test if the current time plus half integration period minus ``bin_low`` is less than some measure. If so, we add the sample. If not we finish the bin and create a new one.

In the UVW averaging case, the test will probably involve checking whether we've travelled far enough in UVW space in order to create a new bin. If we haven't we can safely add the current sample to the bin.